src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -174,7 +175,13 @@ async def _handle_unary_stream_rpc(object method_handler,
 
         # Consumes messages from the generator
         async for response_message in async_response_generator:
-            await servicer_context.write(response_message)
+            if rpc_state.server._status == AIO_SERVER_STATUS_STOPPED:
+                # The async generator might yield much much later after the
+                # server is destroied. If we proceed, Core will crash badly.
+                _LOGGER.warn('RPC Aborted: Server already stopped.')
```

Comment:
Nit: "Aborting RPC due to server stop."

Reply:
I think `warn` is too high a logging level for this behavior, considering it's a documented part of the API. I think `info` or even `debug` would be more appropriate.

> If a grace period is specified, this method returns immediately and all RPCs active at the end of the grace period are aborted. If a grace period is not specified (by passing None for grace), all existing RPCs are aborted immediately and this method blocks until the last RPC handler terminates.